src/js/angular/core/directives/autocomplete/autocomplete.directive.js

```diff
@@ -160,6 +167,42 @@ function autocomplete($location, toastr, ClassInstanceDetailsService, Autocomple
             $scope.activeSearchElm = index;
         };
 
+        /**
+         * Adjusts the position of the autocomplete dropdown based on the viewport's width.
+         *
+         * This function calculates the position of the `.autocomplete-results-wrapper` element
+         * relative to the body of the document. If the right edge of the dropdown extends
+         * beyond the width of the viewport, it adjusts the CSS to align the dropdown right
+         * edge with the right edge of the input field. Otherwise, it aligns the dropdown left
+         * edge with the left edge of the input field. This ensures that the dropdown is fully
+         * visible within the current viewport.
+         *
+         * The function is wrapped inside a `$timeout` to ensure that the DOM has finished
+         * updating before calculating the positions and applying any adjustments.
+         *
+         * Additionally, an angular watch is set up on `autoCompleteUriResults` to reposition
+         * the dropdown whenever the autocomplete results change.
+         */
+        $scope.positionAutocompleteWrapper = function() {
```

Comment:
This behaves quite strange. When I type in the autocomplete dropdown appears and on each keystroke it starts to shift from left aligned to right which is quite awkward

Reply:
fixed